### PR TITLE
fix: file picking issues

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Fixed few issues found during regression of 3.1.0, all the fixes about file choosing capability.

## Links
- MINI-3027 (a fix to allow multiple file picking by @corycaywood)
- MINI-3033

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 1]
```diff
@@ -54,10 +58,34 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
...
+            else -> {
+                callback?.onReceiveValue(null)
+            }
+        }
+    }
+
+    /**
+     * Can be used when HostApp wants to reset the file choosing operation.
+     */
+    fun reset() {
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:88]: Nice solution!

I think maybe the name could be specific on where this is used. How about something like `onCancel` or `cancelReceivingFiles`?

**Review by corycaywood — APPROVED** [review]
LGTM 👍

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt (hunk 1 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt]: Yeah, I have renamed to `onCancel`.

**Comment by corycaywood** [comment]
@khairul-alam-licon Sorry, I forgot to mention: could you update this to target the `candidate` branch instead?

**Review by minh-rakuten — APPROVED** [review]
👍

**Pull request merged**